### ISSUE TITLE
Stricter right primer-trimming in w32 file?

### ISSUE THREAD

**Issue by peterjc** [peterjc/thapbi-pict#482]
Consider this potential 288bp environmental sequence (most from likely _Globisporangium sylvaticum_), from a public dataset after applying our default pipeline primers (despite the fact the PCR was done with the ITS6/7 primers for the second round so not directly compatible):

```
>e6be63b5cd6fd3e658e51721d231455f
TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACT
```

(The sequences this is cropped from had perfect left primers, but the right primer matched ``GTGAGAACGAAAGTTCTTGC`` with 3 mismatches - see below.)

We currently have 46 rather long sequences in ``2022-07-05_ITS1_Oomycota_w32.fasta`` with this start:

```
$ grep ^TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACT ~/repositories/thapbi-pict/database/2022-07-05_ITS1_Oomycota_w32.fasta  -c
46
```

The region of interest is always followed by the same 20bp:

```
$ grep ^TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACT ~/repositories/thapbi-pict/database/2022-07-05_ITS1_Oomycota_w32.fasta | cut -c 289-308 | sort | uniq -c
  46 GTGAGAACGAAAGTTCTTGC
```

We are looking for ``GYRGGGACGAAAGTCYYTGC`` (the RC of 5.8S-1R primer ``GCARRGACTTTCGTCCCYRC`` from Scibetta et al. 2012), which is close but has 3bp different:

```
GTGAGAACGAAAGTTCTTGC - observed
|::X|X||||||||X::|||
GYRGGGACGAAAGTCYYTGC - primer RC
```

From a small sample it looks like these entries in our w32 file have not been right-trimmed at all, e.g.

```
>MK795059.1 Globisporangium sylvaticum isolate VAL1705_160 small subunit ribosomal RNA gene, partial sequence; internal transcribed spacer 1, 5.8S ribosomal RNA gene, and internal transcribed spacer 2, complete sequence; and large subunit ribosomal RNA gene, partial sequence
TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACTGTGAGAACGAAAGTTCTTGCTTTTAACTAGATAACAACTTTCAGCAGTGGATGTCTAGGCTCGCACATCGATGAAGAACGCTGCGAACTGCGATACGTAATGCGAATTGCAGAATTCAGTGAGTCATCGAAATTTTGAACGCATATTGCACTTCCGGGTTATGCCTGGAAGTATGTCTGTATCAGTGTCCGTAAATCAAACTTGCCTTTCTTCCTTCCGTGTAGTCGGTGGAGGAGAGTTGCAGATGTGAAGTGTCTCGCTGTGGTTGGTATATTTGTTTATGCACAACTTGCGAGTCCTTTTAAATGGACACGACTTTCTCTTTTTTGTATCTGCGCGGTGCTGTGCGTGAACGCGGTGGTTTTCGGATCGCTCGCGGCTGTCGGCGACTTCGGTGAATGCATTATGGAGTGGACCTCGATTCGCGGTATGTTGGGCTTCGGCTGGACAATGTTGCTTATTGGGTGTCTGTTCCGCGTTCGCCTTGAGGTGTACTGGTGGCTGTGGGATTGAACTGGTTACTGTTGTTAGTAGTGTGCGGCTCGTTGTCGTGGATGCATCTGTGTTTTTGTACTTGTGCAATTGGCAGAAGAGGAGTCTCAATTTGGGAAATTAGTGTACTCCGGGTTTGATCCTGCGTGTATATCTCAATTGGACCTGATATCAGACAAGACTACCCGCTGAACTTAAGCATATCAT
>KF181455.1 Pythium sylvaticum clone Py3-2 18S ribosomal RNA gene, partial sequence; internal transcribed spacer 1, complete sequence; and 5.8S ribosomal RNA gene, partial sequence
TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACTGTGAGAACGAAAGTTCTTGCTTTTAACTAGATAACA
```

The vast majority fail the maximum length check and are not imported into the DB, with one exception:

```
$ thapbi_pict dump | grep TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACT
ITS1	KF181455.1	Globisporangium		1448052	f6c1630312c013b6dea0263b9364ea09	TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACTGTGAGAACGAAAGTTCTTGCTTTTAACTAGATAACA
```

This too was untrimmed on the right, but by chance the published sequence ended early enough that the fragment was not too large.

Perhaps we should discard all these untrimmed entries (including ``KF181455.1``), and/or relax the matching (which seem reasonable for this example, although we don't know if it would amplify in vitro).

I suspect the reason cutadapt accepted ``GTGAGAACGAAAGTTCTTGC`` with 3bp differences in the pipeline is it was a combined search with the left primer which matched perfectly, and a pooled error limit, whereas when building ``2022-07-05_ITS1_Oomycota_w32.fasta`` the primer searchers were done separately - and thus with their own error limits.

**Comment by peterjc** [comment]
Note that PR #485 right now would not import any of the sequences above containing:

```
>e6be63b5cd6fd3e658e51721d231455f
TTTCCGTAGGTGAACCTGCGGAAGGATCATTACCACACCTAAAAAACTTTCCACGTGAACTGTCGTTATTTGTTGTGTGTCTGCGCGTTGCTGGCGTGCGTTTGCTTACGCTTCGGTGTTTGCGAGTGCGTGCTGGCGGTGCGCGGACTGAACGAAGGTCGTGTTGCTGTGTGCCTGCTGCACTGCTGACTTTGCATTGATTTGCATGGTCTTGGCGGAGCGGCGGGTGCTGTGCGTGCGCGGCTGACTTACTTTTTCAAACCCCATACCTAACTTACTGATTATACT
```

It does however import lots of similar _Globisporangium sylvaticum_ sequences, including just 1bp away.

**Comment by peterjc** [comment]
Marking as closed with #485 now merged for the v0.12.6 release.
